code/modules/mob/living/simple_animal/hostile/megafauna/hivetechboss.dm

```diff
@@ -0,0 +1,101 @@
+/mob/living/simple_animal/hostile/megafauna/hivemind_tyrant
+	name = "Hivemind Tyrant"
+	desc = "Hivemind's will, manifested in flesh and metal."
+
+	faction = "hive"
+
+	icon = 'icons/mob/64x64.dmi'
+	icon_state = "hivemind_tyrant"
+	icon_living = "hivemind_tyrant"
+	icon_dead = "hivemind_tyrant"
+	pixel_x = -16
+	ranged = TRUE
+
+	health = 2500
+	maxHealth = 2500
+	break_stuff_probability = 95
+
+	melee_damage_lower = 10
+	melee_damage_upper = 20
+	megafauna_min_cooldown = 50
+	megafauna_max_cooldown = 80
+
+
+	var/health_marker_1 = 0//1700
+	var/health_marker_2 = 0//900
+	var/health_marker_3 = 0//100
+
+	projectiletype = /obj/item/projectile/goo
+
+/mob/living/simple_animal/hostile/megafauna/hivemind_tyrant/death()
+	..()
+	delhivetech()
+	walk(src, 0)
+
+/mob/living/simple_animal/hostile/megafauna/hivemind_tyrant/proc/telenode()
+	var/list/atom/NODES = list()
+	for(var/obj/machinery/hivemind_machine/node/NODE in world)
+		NODES.Add(NODE.loc)
+	if(length(NODES) > 0)
+		loc = pick(NODES)
```

Comment:
`forceMove(pick(NODES))`
direct setting loc will break stuff like observer events, forceMove must be used so all required stuff happens